haralyzer/http.py

```diff
@@ -0,0 +1,146 @@
+"""Creates the Request and Response sub class that are used by each entry"""
+from cached_property import cached_property
+from .mixins import GetHeaders, MimicDict
+
+
+class Request(GetHeaders, MimicDict, object):
+    """Request object for an HarEntry"""
+    def __init__(self, entry):
+        self.raw_entry = entry
+
+    def __str__(self):
+        return "HarEntry.Request for %s" % self.raw_entry["url"]
+
+    def __repr__(self):
+        return "HarEntry.Request for %s" % self.raw_entry["url"]
+
+    # Root Level values
+
+    @cached_property
+    def bodySize(self):
+        return self.raw_entry["bodySize"]
+
+    @cached_property
+    def cookies(self):
+        return self.raw_entry["cookies"]
+
+    @cached_property
+    def headers(self):
+        return self.raw_entry["headers"]
+
+    @cached_property
+    def headersSize(self):
+        return self.raw_entry["headersSize"]
+
+    @cached_property
+    def httpVersion(self):
+        return self.raw_entry["httpVersion"]
+
+    @cached_property
+    def method(self):
+        return self.raw_entry["method"]
+
+    @cached_property
+    def queryString(self):
+        return self.raw_entry["queryString"]
+
+    @cached_property
+    def url(self):
+        return self.raw_entry["url"]
+
+    # Header Values
+
+    @cached_property
+    def accept(self):
+        return self.get_header_value("Accept")
+
+    @cached_property
+    def cacheControl(self):
+        return self.get_header_value("Cache-Control")
+
+    @cached_property
+    def encoding(self):
+        return self.get_header_value("Accept-Encoding")
+
+    @cached_property
+    def host(self):
+        return self.get_header_value("Host")
+
+    @cached_property
+    def language(self):
+        return self.get_header_value("Accept-Language")
+
+    @cached_property
+    def userAgent(self):
+        return self.get_header_value("User-Agent")
+
+
+class Response(GetHeaders, MimicDict, object):
+    """Response object for a HarEntry"""
+    def __init__(self, entry):
+        self.raw_entry = entry
+
+    # Root Level values
+
+    @cached_property
+    def bodySize(self):
+        return self.raw_entry["bodySize"]
+
+    @cached_property
+    def headers(self):

```

Comment:
There is some repetition between the properties. I think you could have a base class for these, and then add unique properties, something along the lines of:

```
class HttpTransaction(GetHeaders, MimicDict, object):
    def __init__(self, entry):
        self.raw_entry = entry

    # Base class gets properties that belong to both request/response
    @cached_property
    def headers(self):
        return self.raw_entry["headers"]

# Inherit from the base class
class Response(HttpTransaction):

    # Only responses have a redirect URL
    def redirect_url(self):
    ...........
```